build/docker-image/self-signer-cert-utility/Dockerfile

```diff
@@ -18,13 +18,16 @@ COPY cmd/ cmd/
 COPY pkg/ pkg/
 
 # Build the binary self-signer utility
-RUN go build -a -installsuffix cgo -o self-signer cmd/main.go
+RUN go build -o self-signer cmd/main.go
 
 # Install the cockroach CLI
-ADD https://binaries.cockroachdb.com/cockroach-v23.1.2.linux-amd64.tgz .
-RUN tar -zxvf cockroach-v23.1.2.linux-amd64.tgz
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then GOARCH=amd64; elif [ "$TARGETPLATFORM" = "linux/arm64" ]; then GOARCH=arm64; else GOARCH=amd64; fi && \
+    curl -sS -L -O https://binaries.cockroachdb.com/cockroach-v23.1.2.linux-${GOARCH}.tgz && \
```

Comment:
A nit. Can you update the crdb version to the latest one? As a bonus you can move it to a variable which can be passed in. Not a blocker though, feel free to do this as a separate PR.

Reply:
I did even better, now the self-signer automatically picks up the latest cockroach binary version defined in the Chart.yaml. Now, no need for special code changes to upgrade cockroach binary in docker image. Please have a look.